### PR TITLE
add preview controller tests

### PR DESCRIPTION
**Pull request by mamccrea**
#489

### CODE REVIEW

**Comment by lamebrain** [comment]
@mamccrea Hi Mary, same with this PR, looks like there are some conflicts which are causing some test failures. Can you take a look?

File: docs-markdown/src/test/suite/controllers/preview-controller.test.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	});
+	test('noActiveEditorMessage', () => {
+		const spy = chai.spy.on(common, 'noActiveEditorMessage');
+		previewTopic();
+		expect(spy).to.have.been.called();
+	});
+	test('isMarkdownFileCheck', async () => {
+		const filePath = resolve(__dirname, '../../../../../src/test/data/repo/articles/preview.md');
+		await loadDocumentAndGetItReady(filePath);
+		const stub = sinon.stub(telemetry, 'sendTelemetryData');
```
Comment (bharney0) [inline docs-markdown/src/test/suite/controllers/preview-controller.test.ts:36]: can you put this line in a suiteSetup similar to this test https://github.com/microsoft/vscode-docs-authoring/blob/d9f16d58c84427179d24efde462ffac9658a03f4/docs-markdown/src/test/suite/controllers/image-controller.test.ts#L40. You will also need the suiteTearDown https://github.com/microsoft/vscode-docs-authoring/blob/d9f16d58c84427179d24efde462ffac9658a03f4/docs-markdown/src/test/suite/controllers/image-controller.test.ts#L35

File: docs-markdown/src/test/suite/controllers/preview-controller.test.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		previewTopic();
+		expect(spy).to.have.been.called();
+	});
+	test('showPreviewToSide - checkExtension', async () => {
+		const spy = chai.spy.on(common, 'checkExtension');
+		previewTopic();
+		await sleep(100);
+		expect(spy).to.have.been.called();
+	});
+	test('seoPreview - checkExtension', async () => {
```
Comment (bharney0) [inline docs-markdown/src/test/suite/controllers/preview-controller.test.ts:50]: This looks like the same test as above. Did you mean to do seoPreview?

Comment (mamccrea) [reply docs-markdown/src/test/suite/controllers/preview-controller.test.ts:50]: I agree - this is leftover from when I was using the other extension in the test. I'll take it out.

**Comment by mamccrea** [comment]
@lamebrain - it says it's failing on the "Common getOSPlatform returns osPlatform" test, but I'm not sure why/how my tests would impact that test.

**Comment by bharney0** [comment]
> @lamebrain - it says it's failing on the "Common getOSPlatform returns osPlatform" test, but I'm not sure why/how my tests would impact that test.

Its likely that the existing call to `getOSPlatform` is interfering with the test. You may need to stub that call and restore it after the test executes.

**Pull request merged**